cmd/system-probe/common/common.go

```diff
@@ -0,0 +1,27 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package common
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/DataDog/datadog-agent/cmd/system-probe/utils"
+)
+
+var (
+	// MemoryMonitor is the global system-probe memory monitor
+	MemoryMonitor *utils.MemoryMonitor
+
+	// ExpvarServer is the global expvar server
+	ExpvarServer *http.Server
```

Comment:
`stopSystemProbe` is called without any context on windows. All these variables mirror what `agent` does.

Reply:
To clarify, there are two ways into `startSystemProbe` and `stopSystemProbe`, one for Linux and one for Windows. They don't share much code beforehand, so a global was the simpler way to share these variables between `startSystemProbe` and `stopSystemProbe`.

Reply:
If we can fix this now, I think we should. Removing globals where possible is definitely desirable, as would be a more consistent interface between windows and linux. If it's too much trouble it can be addressed in follow-up, but this should be eventually changed.

Please note that expvars and telemetry will eventually have their own component that will be injected, so this is certainly something that will have to be changed in the future.

Reply:
I think a follow up the addresses this issue in a common way could be applied to both agent and system-probe (and any other agent binaries that run on windows). Addressing it here would muddy the waters.

Reply:
windows-agent has an open card to de-duplicate all the Windows service code
https://datadoghq.atlassian.net/browse/WA-163

Reply:
For reference, see https://github.com/DataDog/datadog-agent/pull/15283/commits/3750d442cdc755368e12ec99bf8d48caafdd8539 and https://github.com/DataDog/datadog-agent/pull/15283/commits/1a9b34c173919a095bc7243421b18deb59fcddc5 for how it was done in the core Agent.